Libplanet/Tx/Transaction.cs

```diff
@@ -602,6 +602,12 @@ internal RawTransaction ToRawTransaction(bool includeSign)
         private static T ToAction(IDictionary<string, object> arg)
         {
             var typeStr = (string)arg["type_id"];
+            if (typeStr is null)
+            {
+                throw new InvalidActionTypeException(
+                    "Action type should be annotated with ActionTypeAttribute");
```

Comment:
It's more likely to be a serialization problem rather than the lack of attribute annotation.  It would be better to throw this exception when an action is serialized.

And coul you insert a period to the end of the statement?